### DISCUSSION TITLE
How to connect a local account / smart wallet account?

### DISCUSSION THREAD

**Discussion by stephancill** [wevm/wagmi#4293] [Q&A] [answered]
I would like to use wagmi for my wallet app which has access to a local account via `toCoinbaseSmartAccount`. I essentially want to "connect" this account to wagmi so that I can use it with the hooks.

What is the recommended way of doing this? Do I need to create a connector?

**Comment by tmm** [comment, marked as answer]
Best option right now is to create your own connector with the smart account attached. Wagmi's [`mock`](https://github.com/wevm/wagmi/blob/main/packages/core/src/connectors/mock.ts) connector and [`burner-connector`](https://github.com/scaffold-eth/burner-connector) might be helpful to look at. Longer term, we are exploring adding first-class support to Wagmi.
